visualization/dashboard/src/MLIDashboard/Controls/CohortEditor/CohortEditor.tsx

```diff
@@ -372,14 +417,59 @@ export class CohortEditor extends React.PureComponent<ICohortEditorProps, ICohor
     }
 
     private setFilterLabel(filter: IFilter): React.ReactNode {
-        //TODO: change the function unwrap neatly 
+        const selectedFilter = this.props.jointDataset.metaDict[filter.column];
         let label = "";
-        const space = " ";
-        label = this.props.jointDataset.metaDict[filter.column].abbridgedLabel + space
-        label += filter.method + space
-        label += filter.arg
+        label = selectedFilter.abbridgedLabel
+
+        const filterMethod = this.getFilterMethodLabel(filter.method)
+            label += filterMethod
+        if (selectedFilter.isCategorical) {
+            let selectedValues = [];
+            let filterArgs = filter.arg as number[];
+            filterArgs.forEach((element) => {
```

Comment:
this may need to handle cases where lots of items are selected, and return something like "includes option1, option2, option3, and 58 others..." . You can use Array.slice or Array.splice to take the first few and add the overflow text to the end.

Reply:
Ahh sorry I forgot to implement this based on our conversations!